src/g3log/loglevels.hpp

```diff
@@ -113,13 +144,37 @@ namespace g3 {
 #ifdef G3_DYNAMIC_LOGGING
    // Only safe if done at initialization in a single-thread context
    namespace only_change_at_initialization {
-      // Enable/Disable a log level {DEBUG,INFO,WARNING,FATAL}
+
+      // This sets the logging level status enabled or disabled for a logging level.
+      // If the logging level is NEW it will also ADD the logging level to the set of logging levels.
       void setLogLevel(LEVELS level, bool enabled_status);
+
+      /// Enable log level >= log_level
+      void setLogLevel(LEVELS level);
```

Comment:
`setLogLevel(LEvELS, enabled_status)` means
1. if `LEVEL` did not exist earlier then add it. 
2. set  the enabled flag for the level

`setLogLevel(LEvELS)` means
1. disable all LEVELS below this one
2. enabled all LEVELS from this and higher. 

Since the functionality is completely different a different name should be used. 
`setHighestLogLevel(LEVEL)` makes sense. 

It MIGHT make sense to have a `addLogLevel(LEVEL)` just to make the API  more readable. it would really just be a func forward to` setLogLevel(LEVEL, true)`